From 088101b2e97880e39c2fc3e50765d28a0576babc Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Tue, 5 Nov 2019 08:46:59 +0800
Subject: [PATCH] soc: rockchip: system_monitor: fix clang warning

drivers/soc/rockchip/rockchip_system_monitor.c:579:2: warning:
Attempt to free released memory

Change-Id: I7bba708b6457fed0553a56ec7f943311a490fbb8
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/soc/rockchip/rockchip_system_monitor.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/soc/rockchip/rockchip_system_monitor.c b/drivers/soc/rockchip/rockchip_system_monitor.c
index c5c2668e0c44..9d8ccec6afb4 100644
--- a/drivers/soc/rockchip/rockchip_system_monitor.c
+++ b/drivers/soc/rockchip/rockchip_system_monitor.c
@@ -466,6 +466,7 @@ static int rockchip_init_temp_opp_table(struct monitor_dev_info *info)
 	opp_table = dev_pm_opp_get_opp_table(dev);
 	if (!opp_table) {
 		kfree(info->opp_table);
+		info->opp_table = NULL;
 		return -ENOMEM;
 	}
 	mutex_lock(&opp_table->lock);
-- 
2.35.3

